src/OrchardCore.Modules/OrchardCore.Users/Workflows/Drivers/RegisterUserTaskDisplayDriver.cs

```diff
@@ -10,15 +10,23 @@ public class RegisterUserTaskDisplayDriver : ActivityDisplayDriver<RegisterUserT
         protected override void EditActivity(RegisterUserTask activity, RegisterUserTaskViewModel model)
         {
             model.SendConfirmationEmail = activity.SendConfirmationEmail;
-            model.ConfirmationEmailSubject = activity.ConfirmationEmailSubject.Expression;
-            model.ConfirmationEmailTemplate = activity.ConfirmationEmailTemplate.Expression;
+
+            if (activity.SendConfirmationEmail)
+            {
+                model.ConfirmationEmailSubject = activity.ConfirmationEmailSubject.Expression;
+                model.ConfirmationEmailTemplate = activity.ConfirmationEmailTemplate.Expression;
+            }
         }
 
         protected override void UpdateActivity(RegisterUserTaskViewModel model, RegisterUserTask activity)
         {
             activity.SendConfirmationEmail = model.SendConfirmationEmail;
-            activity.ConfirmationEmailSubject = new WorkflowExpression<string>(model.ConfirmationEmailSubject);
-            activity.ConfirmationEmailTemplate = new WorkflowExpression<string>(model.ConfirmationEmailTemplate);
+
+            if (model.SendConfirmationEmail)
+            {
+                activity.ConfirmationEmailSubject = new WorkflowExpression<string>(model.ConfirmationEmailSubject);
```

Comment:
This isn't validation.

Here is an example of how to add validation to a driver
https://github.com/OrchardCMS/OrchardCore/blob/af6e3313f7a650ea3e2c4931218fb622173957b4/src/OrchardCore.Modules/OrchardCore.ContentFields/Drivers/TextFieldDisplayDriver.cs#L55

Reply:
@deanmarcussen if I use the overload that has the `IUpdaterModel` I can't access the `RegisterUserTask`, any idea?

Reply:
`public async override Task<IDisplayResult> UpdateAsync(TActivity model, IUpdateModel updater)`

try this method

Reply:
I already did that but if you check my previous comment the missing part is `RegisterUserTask` which I can't find

Reply:
@hishamco if you take the time to look and see what that method does, you will see it is merely a helper to create the viewmodel, and bind it. Override it, and change the code to something like this

```
        public async override Task<IDisplayResult> UpdateAsync(TActivity model, IUpdateModel updater)
        {
            var viewModel = new TEditViewModel();
            if (await updater.TryUpdateModelAsync(viewModel, Prefix))
            {
                //await UpdateActivityAsync(viewModel, model);
                Do the regular viewmodel -> model mappings here.
                Validate the values here.
             }

            return Edit(model);
        }
```

Reply:
I though there's an overload with both `IUpdateModel` and `RegisterUserTask` but I don't find it, anyhow I updated the PR, but the strange part that the validation error messages don't show up, but when I place validation summary they show up

Reply:
Looks like you're missing the `Prefix` property when adding the error to the model state

can you merge dev as well, and add the new property in

Reply:
I was wonder why it's important